website/content/api-docs/secret/openldap.mdx

```diff
@@ -216,36 +238,257 @@ $ curl \
+where this value is calculated compared with where Vault calculates details of the lease.<br />
+**Format:** 2006-01-02T15:04:05Z07:00 ([RFC3339](https://golang.org/pkg/time/#pkg-constants))
+
+`.IssueTimeSeconds` - The [unix time](https://en.wikipedia.org/wiki/Unix_time) the lease was created. This time may
+be slightly earlier than the associated lease due to where this value is calculated compared with where Vault
+calculates details of the lease.<br />
+**Format:** Integer indicating the number of seconds elapsed since January 1, 1970.
+
+`.ExpirationTime` - The time that the lease is set to expire. This time may be slightly earlier than the associated
+lease due to where this value is calculated compared with where Vault calculates details of the lease.<br />
+**Format:** 2006-01-02T15:04:05Z07:00 ([RFC3339](https://golang.org/pkg/time/#pkg-constants))
+
+`.ExpirationTimeSeconds` - The [unix time](https://en.wikipedia.org/wiki/Unix_time) the lease is set to expire. This
+time may be slightly earlier than the associated lease due to where this value is calculated compared with where Vault
+calculates details of the lease.<br />
+**Format:** Integer indicating the number of seconds elapsed since January 1, 1970.
+
+#### Username Template Fields
+
+The following parameters are available within the username template:
+
+**Important note:** If any of the following fields include dashes or underscores, they will not be removed/changed
+unless explicitly done so within the username template. For instance:
+
+If `RoleName` is `test-role` and the `username_template` is `v_{{.RoleName}}_{{unix_time}}`, the result of this template
+may be: `v_test-role_1234567890`. Note the `-` (dash) in the `test-role`. If the LDAP system Vault is managing
+restricts usernames/DNs to not allow for dashes (or other characters), the template must explicitly modify/remove those
+characters. In this example, the template can be changed to `v_{{.RoleName | replace "-" "_"}}` to replace
+the dashes with underscores. See [Template Functions](#template-functions) for more functions available.
+
+`.RoleName` - The name of the role the credentials are being generated from.
+
+`.DisplayName` - The display name associated with the user making the request against Value.
+
+#### Template Functions
+
+Both the LDIF templates as well as the username template support using the following functions:
+
+`random` - Generates a random string from lowercase letters, uppercase letters, and numbers. Must include a
+number indicating how many characters to generate.<br />
+**Example:** `{{random 20}}` generates 20 random characters
+
+`truncate` - truncates the previous value to the provided number of characters.<br />
+**Example:** `{{.FieldName | truncate 10}}`
+
+`truncate_sha256` - Truncates the previous value to the provided number of characters. The last 8 characters of the
```

Comment:
It might be helpful to describe when this function is useful.

Reply:
Updated to have more details. I also included a more concrete scenario example. Let me know how this reads.